h/api/models/test/document_test.py

```diff
@@ -434,6 +435,203 @@ def merge_data(self, request):
         return (master, duplicate)
 
 
+class TestUpdateDocumentMetadata(object):
+
+    def test_it_uses_the_target_uri_to_get_the_document(self,
+                                                        annotation,
+                                                        Document,
+                                                        session):
+        document_uri_dicts = [
+            {
+                'uri': 'http://example.com/example_1',
+                'claimant': 'http://example.com/claimant',
+                'type': 'type',
+                'content_type': None,
+            },
+            {
+                'uri': 'http://example.com/example_2',
+                'claimant': 'http://example.com/claimant',
+                'type': 'type',
+                'content_type': None,
+            },
+            {
+                'uri': 'http://example.com/example_3',
+                'claimant': 'http://example.com/claimant',
+                'type': 'type',
+                'content_type': None,
+            },
+        ]
+
+        document.update_document_metadata(session,
+                                          annotation,
+                                          [],
+                                          document_uri_dicts)
+
+        Document.find_or_create_by_uris.assert_called_once_with(
+            session,
+            annotation.target_uri,
+            [
+                'http://example.com/example_1',
+                'http://example.com/example_2',
+                'http://example.com/example_3',
+            ],
+            created=annotation.created,
+            updated=annotation.updated,
+        )
+
+    def test_if_there_are_multiple_documents_it_merges_them_into_one(
+            self,
+            annotation,
+            Document,
+            merge_documents,
+            session):
+        """If it finds more than one document it calls merge_documents()."""
+        Document.find_or_create_by_uris.return_value = mock.Mock(
+            count=mock.Mock(return_value=3))
+
+        document.update_document_metadata(session, annotation, [], [])
+
+        merge_documents.assert_called_once_with(
+            session,
+            Document.find_or_create_by_uris.return_value,
+            updated=annotation.updated)
+
+    def test_it_calls_first(self, annotation, session, Document):
+        """If it finds only one document it calls first()."""
+        Document.find_or_create_by_uris.return_value = mock.Mock(
+            count=mock.Mock(return_value=1))
+
+        document.update_document_metadata(session, annotation, [], [])
+
+        Document.find_or_create_by_uris.return_value\
+            .first.assert_called_once_with()
```

Comment:
Oh, except that `update_document_metadata()` isn't in `storage` anymore because it moved into `models`!

So in that case yes, probably it should not mock the db (the reason it does is that this function these tests were in `storage` and then they moved into `models` and I didn't rewrite the tests)